DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/CourseSetup/CourseSetupControllerTests.cs

```diff
@@ -0,0 +1,152 @@
+﻿namespace DigitalLearningSolutions.Web.Tests.Controllers.TrackingSystem.CourseSetup
+{
+    using System.Collections.Generic;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Models.Common;
+    using DigitalLearningSolutions.Data.Models.Courses;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Controllers.TrackingSystem.CourseSetup;
+    using DigitalLearningSolutions.Web.Tests.ControllerHelpers;
+    using DigitalLearningSolutions.Web.ViewModels.TrackingSystem.CourseSetup;
+    using FakeItEasy;
+    using FluentAssertions;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using NUnit.Framework;
+
+    public class CourseSetupControllerTests
+    {
+        private readonly List<Category> categories = new List<Category>
+        {
+            new Category { CategoryName = "Category 1" },
+            new Category { CategoryName = "Category 2" }
+        };
+
+        private readonly List<CourseStatistics> courses = new List<CourseStatistics>
+        {
+            new CourseStatistics
+            {
+                ApplicationName = "Course",
+                CustomisationName = "Customisation",
+                Active = true,
+                CourseTopic = "Topic 1",
+                CategoryName = "Category 1",
+                HideInLearnerPortal = true,
+                DelegateCount = 1,
+                CompletedCount = 1
+            }
+        };
+
+        private readonly List<Topic> topics = new List<Topic>
+        {
+            new Topic { CourseTopic = "Topic 1" },
+            new Topic { CourseTopic = "Topic 2" }
+        };
+
+        private CourseSetupController controller = null!;
+        private ICourseCategoriesDataService courseCategoryDataService = null!;
+        private ICourseService courseService = null!;
+        private ICourseTopicsDataService courseTopicsDataService = null!;
+        private HttpRequest httpRequest = null!;
+        private HttpResponse httpResponse = null!;
+
+        [SetUp]
+        public void Setup()
+        {
+            courseCategoryDataService = A.Fake<ICourseCategoriesDataService>();
+            courseTopicsDataService = A.Fake<ICourseTopicsDataService>();
+            courseService = A.Fake<ICourseService>();
+
+            A.CallTo(() => courseService.GetCentreSpecificCourseStatistics(A<int>._, A<int>._)).Returns(courses);
+            A.CallTo(() => courseCategoryDataService.GetCategoriesForCentreAndCentrallyManagedCourses(A<int>._))
+                .Returns(categories);
+            A.CallTo(() => courseTopicsDataService.GetCourseTopicsAvailableAtCentre(A<int>._)).Returns(topics);
+
+            httpRequest = A.Fake<HttpRequest>();
+            httpResponse = A.Fake<HttpResponse>();
+            const string cookieName = "CourseFilter";
+            const string cookieValue = "Status|Active|true";

```

Comment:
I feel like this cookie value is not an optimal choice as we will probably want a test for the case of "no cookie, status is set to active" case. It would be good if the filter in the cookie is different from that test case.